frame/nomination-pools/src/lib.rs

```diff
@@ -2328,20 +2667,45 @@ impl<T: Config> Pallet<T> {
 		member.last_recorded_reward_counter = current_reward_counter;
 		reward_pool.register_claimed_reward(pending_rewards);
 
-		// Transfer payout to the member.
-		T::Currency::transfer(
-			&bonded_pool.reward_account(),
-			&member_account,
-			pending_rewards,
-			// defensive: the depositor has put existential deposit into the pool and it stays
-			// untouched, reward account shall not die.
-			ExistenceRequirement::AllowDeath,
-		)?;
+		// Gets the commission percentage and payee to be paid if commission has been set.
+		// Otherwise, `None` is returned.
+		let maybe_commission = &bonded_pool.commission.maybe_commission_and_payee(&pending_rewards);
+
+		if let Some((pool_commission, payee)) = maybe_commission {
+			// Deduct any outstanding commission from the reward being claimed.
+			pending_rewards = pending_rewards.saturating_sub(*pool_commission);
+
+			// Send any non-zero `pool_commission` to the commission `payee`.
+			if pool_commission > &Zero::zero() {
+				T::Currency::transfer(
+					&bonded_pool.reward_account(),
+					&payee,
+					*pool_commission,
+					ExistenceRequirement::KeepAlive,
+				)?;
```

Comment:
Not sure how to handle the dust case here, when the commission is lower than the ED it will not be able to endow an account.  
So you can probably ignore a dust error here and just require the payee to exist already.  
Otherwise the pool operator could hinder users from claiming their rewards by setting a non-existing account.

Reply:
So if the requester has less than the ED, and the received commission on top of this will also be below ED, then the requester will lose the payout? 

Would the requester be required to do the keep alive checks on their end to prevent this?